pygmt/src/legend.py

```diff
@@ -75,12 +87,11 @@ def legend(self, spec=None, position="JTR+jTR+o0.2c", box="+gwhite+p1p", **kwarg
         if kwargs.get("F") is None:
             kwargs["F"] = box
 
+    kind = data_kind(spec)
+    if kind not in {"vectors", "file"}:  # kind="vectors" means spec is None
```

Comment:
This line can also be written as:
```
if spec is not None or kind != "file"
```
It's written in the current way so that this line can be extended to check "stringio" kind with minor changes (xref: https://github.com/GenericMappingTools/pygmt/pull/3326).

It's not straightforward to see why `kind="vectors"` means `spec is None`. That's also why I want to refactor the `data_kind` function in  #3351.